app/src/main/java/com/tari/android/wallet/ui/fragment/send/finalize/YatFinalizeSendTxActivity.kt

```diff
@@ -22,17 +26,26 @@ class YatFinalizeSendTxActivity : YatLibOutcomingTransactionActivity() {
 
         viewModel.transactionData = entity
         subscribeOnUI()
+        launchObserver()
     }
 
     private fun subscribeOnUI() = with(viewModel) {
         observe(txFailureReason) { setTransactionState(TransactionState.Failed) }
 
-        observe(torConnected) { viewModel.sendTari() }
-
-        observe(finishedSending) { setTransactionState(TransactionState.Complete) }
+        observe(isSuccess) { setTransactionState(TransactionState.Complete) }
 
         observeOnLoad(sentTxId)
-        observeOnLoad(currentStep)
+        observeOnLoad(steps)
+        observeOnLoad(nextStep)
+    }
+
+    private fun launchObserver() {
+        viewModel.start()
+
+        lifecycleScope.launch(Dispatchers.IO) {
+            delay(200)
```

Comment:
delay?

Reply:
It's an imitation of toggling UI as on AddAmountFragment. Prefer to stay this way because it's a low-cost solution that affects nothing in this case. On the other side writing complex code by linking steps on FinalizeSendTxViemModel which needs to be extra testing. Basically just using the existing solution for different UI